web-registry/src/types/ledger/ecocredit.ts

```diff
@@ -2,37 +2,52 @@ import type { PageResponse } from './base';
 import type { QueryBalanceResponse as BankQueryBalanceResponse } from '@regen-network/api/lib/generated/cosmos/bank/v1beta1/query';
 
 // Response structure based on https://buf.build/regen/regen-ledger
-export interface BatchRowData {
+export interface BatchData {
   start_date: string | Date;
   end_date: string | Date;
   issuer: string;
   batch_denom: string;
   class_id: string;
   total_amount: number;
-  tradable_supply?: number;
-  retired_supply?: number;
   amount_cancelled: number;
   project_location: string;
 }
 
-export interface BatchDataResponse {
-  data: BatchRowData[];
-  pagination?: PageResponse;
-}
+export interface BatchRowData extends BatchData, BatchSupplyResponse {}
 
-export interface TableCredits extends BatchRowData, QueryBalanceResponse {}
+export interface TableCredits extends BatchData, QueryBalanceResponse {}
 
 export interface TableBaskets extends Basket, BankQueryBalanceResponse {
   display_denom: string;
 }
 
 // The following interfaces should be removed once we migrate
 // the current queries to use regen-js instead of REST
+
+export interface BatchTotalsForProject {
+  amount_cancelled: number;
+  tradable_supply: number;
+  retired_supply: number;
+}
+
+export interface BatchSupplyResponse {
+  tradable_supply: number;
+  retired_supply: number;
+}
+
+export interface BatchDataResponse {
+  data: BatchRowData[];
+  pagination?: PageResponse;
+}
```

Comment:
the interfaces below the comment "The following interfaces should be removed once we migrate the current queries to use regen-js instead of REST" should be importable from regen-js and have consistent names with the ones on regen-ledger which doesn't seem to be the case here, see following comments